### ISSUE TITLE
broken countrycode detection in v1.0.60

### ISSUE THREAD

**Issue by alicebob** [nyaruka/phonenumbers#75]
Hi,

thanks for the useful library!

I got a number which gets parsed correctly in v1.0.59, but not anymore in v1.0.60.

The call is to `Parse("6282240080000", "ID")`, which returns `+626282240080000`, but it should detect that the `62` is already present and make it `+6282240080000`.
The original number is possible and valid according to https://rawgit.com/googlei18n/libphonenumber/master/javascript/i18n/phonenumbers/demo-compiled.html but the new number (`+6262822...`) is possible but not(!) valid.

A shorter version of the number (`62822400800`) does get parsed and formatted correctly.

This shows the problem:
```
diff --git a/phonenumbers_test.go b/phonenumbers_test.go
index a7e16bd..dc58af3 100644
--- a/phonenumbers_test.go
+++ b/phonenumbers_test.go
@@ -1308,6 +1308,9 @@ func TestParsing(t *testing.T) {
                {"+2203693200", "", "+2203693200"},
                {"0877747666", "ID", "+62877747666"},
                {"62816640000", "ID", "+62816640000"},
+
+               {"62822400800", "ID", "+62822400800"},
+               {"6282240080000", "ID", "+6282240080000"},
        }
 
        for _, tc := range tests {
```

**Comment by cristaloleg** [comment]
Hi, sorry for the trouble. Yep, in v1.0.60 behaviour has slightly changed according to original `googlei18n/libphonenumber`. I will check your case and see what is going on. Thank you for the test!

**Comment by cristaloleg** [comment]
Hey, after a hour of debug I've found why it returns number with a country code.

There is a func `testNumberLength` which checks if the number can have a specific length and for one of your examples (`62822400800`) it says `IS_POSSIBLE` (where available lengths are `[7 8 9 10 11 12 13]`). And for the UK number `447531669965` it returns `[7 9 10]` which is `TOO_LONG`.

That's why after https://github.com/nyaruka/phonenumbers/pull/71 with a fixed regex another checks (`testNumberLength`) is executed. 

Currently I don't have a fix, I would like to test it against `googlei18n/libphonenumber` and see it's behaviour.

**Comment by nicpottier** [comment]
This should be fixed in v1.0.62, apologies for the trouble!

**Comment by alicebob** [comment]
Thanks!
